### PR TITLE
feat(services): move auto-deploy into service settings

### PR DESCRIPTION
**Pull request by ctjhoa**
# What does this PR do?

https://qovery.atlassian.net/browse/FRT-668

2023-09-20-165749_1386x944_scrot
2023-09-20-165832_1120x729_scrot

### CODE REVIEW

**Review by RemiBonnet — CHANGES_REQUESTED** [review]

File: libs/pages/application/src/lib/feature/page-settings-general-feature/page-settings-general-feature.tsx [hunk 1]
```diff
@@ -52,6 +53,7 @@ export const handleContainerSubmit = (data: FieldValues, application: Applicatio
     ...application,
     name: data['name'],
     description: data['description'] || '',
+    ['auto_deploy']: data['auto_deploy'],
```
[the remark this answers is not in the document]
Comment (ctjhoa) [reply libs/pages/application/src/lib/feature/page-settings-general-feature/page-settings-general-feature.tsx]: https://github.com/Qovery/console/pull/843/commits/1408d4e662854dbdc54ffa51118e414e560bfcda

File: libs/pages/application/src/lib/feature/page-settings-general-feature/page-settings-general-feature.tsx [hunk 2]
```diff
@@ -72,6 +74,7 @@ export const handleJobSubmit = (data: FieldValues, application: ApplicationEntit
       ...application,
       name: data['name'],
       description: data['description'],
+      ['auto_deploy']: data['auto_deploy'],
```
[the remark this answers is not in the document]
Comment (ctjhoa) [reply libs/pages/application/src/lib/feature/page-settings-general-feature/page-settings-general-feature.tsx]: https://github.com/Qovery/console/pull/843/commits/1408d4e662854dbdc54ffa51118e414e560bfcda

File: libs/pages/application/src/lib/feature/page-settings-general-feature/page-settings-general-feature.tsx [hunk 3]
```diff
@@ -84,6 +87,7 @@ export const handleJobSubmit = (data: FieldValues, application: ApplicationEntit
...
```
[the remark this answers is not in the document]
Comment (ctjhoa) [reply libs/pages/application/src/lib/feature/page-settings-general-feature/page-settings-general-feature.tsx]: https://github.com/Qovery/console/pull/843/commits/1408d4e662854dbdc54ffa51118e414e560bfcda

File: libs/pages/services/src/lib/ui/page-application-create/step-general/step-general.tsx [hunk 4]
```diff
@@ -100,9 +109,26 @@ export function StepGeneral(props: StepGeneralProps) {
 
         {isContainer(watchServiceType) && <GeneralContainerSettings organization={props.organization} />}
 
-        {watchBuildMode === BuildModeEnum.DOCKER && <EntrypointCmdInputs />}
```
[the remark this answers is not in the document]
Comment (ctjhoa) [reply libs/pages/services/src/lib/ui/page-application-create/step-general/step-general.tsx:103]: https://github.com/Qovery/console/pull/843/commits/b01decfded2b3201442268369cf70cd1507c84ae

**Review by RemiBonnet — CHANGES_REQUESTED** [review]

File: libs/pages/services/src/lib/ui/page-application-create/step-general/step-general.tsx [hunk 5]
```diff
@@ -102,7 +111,26 @@ export function StepGeneral(props: StepGeneralProps) {
 
         {watchBuildMode === BuildModeEnum.DOCKER && <EntrypointCmdInputs />}
 
-        <div className="flex justify-between">
+        <Controller
```
[the remark this answers is not in the document]
Comment (ctjhoa) [reply libs/pages/services/src/lib/ui/page-application-create/step-general/step-general.tsx]: Done https://github.com/Qovery/console/pull/843/commits/322554d9c41e1a9b98283e395111a02550f9d2a1

File: libs/pages/application/src/lib/ui/page-settings-general/page-settings-general.tsx [hunk 6]
```diff
@@ -77,21 +83,20 @@ export function PageSettingsGeneral(props: PageSettingsGeneralProps) {
...
               jobType={isCronJob(type) ? ServiceTypeEnum.CRON_JOB : ServiceTypeEnum.LIFECYCLE_JOB}
-              organization={props.organization}
+              organization={organization}
             />
           )}
           {isApplication(type) && (
             <>
               <EditGitRepositorySettingsFeature />
-              <BlockContent title="Build mode">
+              <BlockContent classNameContent="gap-3 flex flex-col" title="Build & deploy">
```
Comment (RemiBonnet) [inline libs/pages/application/src/lib/ui/page-settings-general/page-settings-general.tsx:92]: should it stay "build mode" no?

**Review by RemiBonnet — APPROVED** [review]
Thanks @ctjhoa, one small comment but after you can merge it !

File: libs/pages/application/src/lib/ui/page-settings-general/page-settings-general.tsx (hunk 6 above)
Comment (ctjhoa) [reply libs/pages/application/src/lib/ui/page-settings-general/page-settings-general.tsx:92]: it should be discuss with @alecarrano when he'll come back from vacation

**Pull request merged**